src/DoctrineORMModule/Form/Annotation/AnnotationBuilder.php

```diff
@@ -5,8 +5,8 @@
 namespace DoctrineORMModule\Form\Annotation;
 
 use ArrayObject;
-use Doctrine\Common\Persistence\Mapping\ClassMetadata;
-use Doctrine\Common\Persistence\ObjectManager;
+use Doctrine\Persistence\Mapping\ClassMetadata;
+use Doctrine\Persistence\ObjectManager;
```

Comment:
Because of the signature ligne 114, and since the class is not `final`, you will need to add a `class_exists(ClassMetadata::class);` to trigger the autoloading as per [my blog post](https://dev.to/greg0ire/how-to-deprecate-a-type-in-php-48cf)

Reply:
@greg0ire what if we only support `doctrine/persistance:^2` we could drop this hack? And `doctrine/doctrine-module:^4.1` already requires persistence:^2

Reply:
@svycka in that case yes, the hack can be dropped.

Reply:
I've dropped v1 persistence support https://github.com/doctrine/DoctrineORMModule/pull/643/commits/ed2bd4364b3df3a6ee4ab8490a2d1ffd4b07aea4

Reply:
I won't be limiting this PR to persistence 2.0 only because this is not the right place for such a change and it would require a new minor version number.  I'll put the hack for `class_exists` in place.

Reply:
@greg0ire your solution has been added